spec/controllers/registrations_controller_spec.rb

```diff
@@ -120,6 +120,96 @@
       end
     end
   end
+
+  describe '#update' do
+    let!(:user) { FactoryGirl.create(:user) }
+
+    context 'valid update' do
+      before(:each) do
+        user.confirm
+        sign_in user
+        patch :update,
+          params: { user: { email: 'new@test.com', password: '123456', password_confirmation: '123456', current_password: user.password } },
+          xhr: true
+        user.reload
+      end
+
+      it 'responds HTTP 200' do
+        expect(response).to have_http_status(200)
+      end
+
+      it 'update email' do
+        expect(user.email).to eq('new@test.com')
+      end
+
+      it 'update password' do
+         expect(user.valid_password?('123456')).to be(true)
```

Comment:
You shouldn't test the validity of the password in you controller spec. I think that this kind of things should be tested on model specs.